### DISCUSSION TITLE
connecting x agent to a local llm using llm studio?

### DISCUSSION THREAD

**Discussion by SmellyBones** [OpenBMB/XAgent#97] [Ideas]
its kind of expensive to use gpt 4 for xagent. im wanting to use my own llm through lm studio. i understand i have to create a server with lm studio. how would i go about connecting xagent to my lm studio server? also if im using a 7b llm model such as something like vicuna or mistral or llama, how much capability am i losing out on? could i have it run off my local llm but keep the gpt api key so if it needs to use gpt for more advanced task it could optionally use gpt.  just an idea but any advice would be appreciated  :D

**Comment by izevar** [comment]
+1 on this topic.
I did try to configure a local LLM without success.
